test/dependencies/URLDependency.unittest.js

```diff
@@ -0,0 +1,212 @@
+							request: "test.js"
+						})
+					},
+					chunkGraph: mockChunkGraph,
+					runtimeRequirements: mockRuntimeRequirements
+				}
+			);
+			
+			// Should only have preload, not prefetch
+			expect(mockRuntimeRequirements.has(RuntimeGlobals.preloadAsset)).toBe(true);
+			expect(mockRuntimeRequirements.has(RuntimeGlobals.prefetchAsset)).toBe(false);
+			
+			const replacementCall = mockSource.replace.mock.calls[0];
+			const replacementCode = replacementCall[2];
+			
+			expect(replacementCode).toContain("__webpack_require__.LA(url, \"script\", \"high\");");
+			expect(replacementCode).not.toContain("__webpack_require__.PA");
+		});
+
+		it("should handle undefined fetchPriority", () => {
+			const dep = new URLDependency(
+				"./test.png",
+				[10, 20],
+				[0, 30],
+				"prefetch"
+			);
+			dep.prefetch = true;
+			// fetchPriority is undefined
+			
+			const template = new URLDependency.Template();
+			template.apply(
+				dep,
+				mockSource,
+				{
+					runtimeTemplate: mockRuntimeTemplate,
+					moduleGraph: mockModuleGraph,
+					chunkGraph: mockChunkGraph,
+					runtimeRequirements: mockRuntimeRequirements
+				}
+			);
+			
+			const replacementCall = mockSource.replace.mock.calls[0];
+			const replacementCode = replacementCall[2];
+			
+			expect(replacementCode).toContain("__webpack_require__.PA(url, \"image\", undefined);");
+		});
+
+		it("should correctly determine asset types", () => {
+			const testCases = [
+				{ request: "test.png", expectedAs: "image" },
+				{ request: "test.jpg", expectedAs: "image" },
+				{ request: "test.webp", expectedAs: "image" },
+				{ request: "test.css", expectedAs: "style" },
+				{ request: "test.js", expectedAs: "script" },
+				{ request: "test.mjs", expectedAs: "script" },
+				{ request: "test.woff2", expectedAs: "font" },
+				{ request: "test.ttf", expectedAs: "font" },
+				{ request: "test.vtt", expectedAs: "track" },
+				{ request: "test.mp4", expectedAs: "fetch" }, // video uses fetch
+				{ request: "test.json", expectedAs: "fetch" },
+				{ request: "test.wasm", expectedAs: "fetch" }
+			];
+			
+			testCases.forEach(({ request, expectedAs }) => {
+				const dep = new URLDependency(
+					`./${request}`,
+					[10, 20],
+					[0, 30],
+					"prefetch"
+				);
+				dep.prefetch = true;
+				dep.fetchPriority = "high";
+				
+				const template = new URLDependency.Template();
+				template.apply(
+					dep,
+					mockSource,
+					{
+						runtimeTemplate: mockRuntimeTemplate,
+						moduleGraph: {
+							getModule: jest.fn().mockReturnValue({ request })
+						},
+						chunkGraph: mockChunkGraph,
+						runtimeRequirements: new Set()
+					}
+				);
+				
+				const replacementCall = mockSource.replace.mock.calls[0];
+				const replacementCode = replacementCall[2];
+				
+				expect(replacementCode).toContain(`__webpack_require__.PA(url, "${expectedAs}", "high");`);
+				
+				mockSource.replace.mockClear();
+			});
+		});
+	});
+});
```

Comment:
Avoid unit tests, they are useless in term of webpack testing and DOM accessing , integration tests take more time but they are more valuable and allow you to catch regressions

Reply:
removed in c632fcc9eef369f2b1f7deddefc464427a748e4b